main_script_HW14.py

```diff
@@ -0,0 +1,101 @@
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+
+    @abstractmethod
+    def __getitem__(self, index) -> str:
+        pass
+
+    @abstractmethod
+    def __str__(self) -> str:
+        pass
+
+    @abstractmethod
+    def __repr__(self) -> str:
+        pass
+
+    @abstractmethod
+    def check_alphabet(self) -> bool:
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, seq: str):
+        self.seq = seq
+        self.length = len(self.seq)
+        self.check_alphabet()
+
+    def __getitem__(self, slc) -> str:
+        if isinstance(slc, int):
+            return self.seq.__getitem__(slc - 1)
+        elif isinstance(slc, slice):
+            new_slice = slice(slc.start - 1, slc.stop, slc.step)
+            return self.seq.__getitem__(new_slice)
+
+    def __str__(self) -> str:
+        return f'{self.seq}'
+
+    def __repr__(self) -> str:
+        return f'{self.seq}'
+
+    def check_alphabet(self) -> bool:
+        unique_char_seq = set(self.seq)
+        return all(nuc in self.complement_dict for nuc in unique_char_seq)
+
+    def complement(self):
+        seq_list = list(self.seq)
+        comp_seq = [self.complement_dict[nuc] for nuc in seq_list]
+        return self.__class__(''.join(comp_seq))
+
+    def gc_content(self) -> float:
+        gc_count = sum(1 for base in self.seq if base.upper() in ['G', 'C'])
```

Comment:
Оригинальный подход! Можно наверное вообще как нибудь типа:
```suggestion
        gc_count = sum(base.upper() in ['G', 'C'] for base in self.seq)
```